pandas/tseries/tools.py

```diff
@@ -283,10 +303,25 @@ def to_datetime(arg, errors='raise', dayfirst=False, yearfirst=False,
     NaT
 
     """
-    return _to_datetime(arg, errors=errors, dayfirst=dayfirst,
-                        yearfirst=yearfirst,
-                        utc=utc, box=box, format=format, exact=exact,
-                        unit=unit, infer_datetime_format=infer_datetime_format)
+    # variable to set offset as per origin parameter
+    offset = None
+
+    if origin == 'julian':
+        if unit != 'D':
+            raise ValueError("unit must be 'D' for origin='julian'")
+        arg = arg - tslib.Timestamp(0).to_julian_date()
```

Comment:
The ``to_julian_date`` method is an example of [Variant 1 with epoch: 12:00PM Jan 1, 4713 BC](https://en.wikipedia.org/wiki/Julian_day#Variants). Thus the desired output for input ``[1,2]`` is ``['4713BC-01-02 12:00PM','4713BC-01-03 12:00PM' ]`` which is out of the the range that could be handled.